secure_tunneling/source/SecureTunnel.cpp

```diff
@@ -728,10 +909,22 @@ namespace Aws
             }
         }
 
-        void SecureTunnel::s_OnSendDataComplete(int error_code, void *user_data)
+        void SecureTunnel::s_OnSendMessageComplete(
+            enum aws_secure_tunnel_message_type type,
+            int error_code,
+            void *user_data)
         {
-            SecureTunnel *secureTunnel = static_cast<SecureTunnel *>(user_data);
-            if (secureTunnel->m_OnSendDataComplete)
+            auto *secureTunnel = static_cast<SecureTunnel *>(user_data);
+
+            if (secureTunnel->m_OnSendMessageComplete)
+            {
+                std::shared_ptr<SendMessageCompleteData> packet =
+                    std::make_shared<SendMessageCompleteData>(type, secureTunnel->m_allocator);
+                SendMessageCompleteEventData eventData;
+                eventData.sendMessageCompleteData = packet;
+                secureTunnel->m_OnSendMessageComplete(secureTunnel, error_code, eventData);
+            }
+            else if (secureTunnel->m_OnSendDataComplete)
```

Comment:
Debatable: should we use `if` instead of `else if` here? Well, technically if the user already sets OnSendMessageComplete, they would not and should not use OnSendDataComplete in this case. I feel it is better to make sure the callbacks work independently?  Or at least mentioned in the comments that if you've set OnSendMessageComplete, OnSendDataComplete would no longer be trigged.